core/src/main/java/org/infinispan/globalstate/impl/GlobalConfigurationStateListener.java

```diff
@@ -42,8 +47,23 @@
             gcm.createTemplateLocally(name, state);
    }
 
+   @CacheEntryModified
+   public CompletionStage<Void> handleUpdate(CacheEntryModifiedEvent<ScopedState, CacheState> event) {
+      String scope = event.getKey().getScope();
+      if (!isKnownScope(scope))
+         return CompletableFutures.completedNull();
+
+      String name = event.getKey().getName();
+      CacheState state = event.getValue();
+
+      return event.isPre() ? gcm.validateConfigurationLocally(name, state) : gcm.updateConfigurationLocally(name, state);
```

Comment:
What happens if validation fails during `event.isPre()`? I assume the `CompletionStage` will complete exceptionally, but won't the post event still be triggered and `updateConfigurationLocally` still be called?

Reply:
According to the `CacheEntryModified` javadoc:
Any exceptions thrown by the listener will abort the call. Any other listeners not yet called will not be called,
and any transactions in progress will be rolled back.

Reply:
:+1: I had  a quick look at the interceptor code and I see how it works now.